lisa/runners/legacy_runner.py

```diff
@@ -405,12 +405,13 @@ def discover_cases(self) -> List[Dict[str, str]]:
         Discover all cases names. The name may be duplicate by test matrix.
         """
         all_cases: List[Dict[str, str]] = []
-        count: int = 1
+        count: int = 0
         for line in self._line_iter():
             case_match = self.CASE_COLLECTED.match(line)
             if case_match:
                 case = {"name": case_match["name"]}
                 all_cases.append(case)
+                count = len(all_cases)
```

Comment:
forget to mention, there may be still a bug. Plase add a TODO. when discover_cases returns results, it implicts all test cases found, so it won't be called again. So if not all test case log produced, and this function is called, some test cases may be lost. If this kind of issue happens, it will fail in later assertions. We will fix it, when it happens.